packages/drivers/odsp-driver/src/createNewUtils.ts

```diff
@@ -210,53 +210,59 @@ export async function createNewFluidContainerCore<T>(args: {
 }): Promise<T> {
 	const {
 		containerSnapshot,
-		getStorageToken,
+		getStorageToken: getAuthHeader,
 		logger,
 		initialUrl,
-		forceAccessTokenViaAuthorizationHeader,
 		epochTracker,
 		telemetryName,
 		fetchType,
 		validateResponseCallback,
 	} = args;
 
 	return getWithRetryForTokenRefresh(async (options) => {
-		const storageToken = await getStorageToken(options, telemetryName);
-
 		return PerformanceEvent.timedExecAsync(
 			logger,
 			{ eventName: telemetryName },
 			async (event) => {
+				// TODO: get rid of the double token fetch in this function, may be we should disable the POST support when authorizationHeader is used?
```

Comment:
What are your thoughts here?

Reply:
Can you please provide a bit more detail? The getWithRetryForTokenRefresh() loop is to account for expired token and the need to refresh it. Are you saying this flow is not required if we are using new code path? Maybe the differences are - we use cached tokens on old path, and new path always gives us fresh token?
If so, it feels like this kind of knowledge should be baked into getWithRetryForTokenRefresh somehow

Reply:
This code dynamically decides whether to include Authorization in the body dynamically based on body length. Because of this we need to request a token assuming it will work, then if it doesn't work, then we need to request a "new" token (because the url changes). This comment was about whether we should do something different (e.g. request a single token assuming Authorization header, then if the token is not a 'Bearer' then disable sending the token in the body.